packages/web-config-server/src/apiV1/dashboardses.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { RouteHandler } from './RouteHandler';
+import { NoPermissionRequiredChecker } from './permissions';
+
+const sortByDbColumns = (array, columns) => {
+  // Sorts an array of objects by properties in a prioritized order
+  let sortedOutput = [];
+  columns.forEach(columnName => {
+    sortedOutput = sortedOutput.concat(array
+      .filter(item => !!item[columnName])
+      .sort((itemA, itemB) => itemA[columnName] > itemB[columnName] ? 1 : -1));
+    array = array.filter(item => !item[columnName]);
+  });
+  return [...sortedOutput, ...array];
+}
```

Comment:
A few things to fix up with this function:

1. Even if you know how it's being used outside the function, you should almost never mutate a function's parameters. It causes bugs because variables change opaquely from the caller's perspective
2. Try to use more descriptive names than `array`. What's in the array? 
3. It's called `sortByDbColumns`, but could work on properties other than db columns, so maybe a more generic name to reflect that
4. I haven't tested the logic properly, but reading through it looks like it filters all with one property defined first, then those with the secondary property defined second and so on, but ideally it would also do conflict resolution using the second property  (I suspect we'll end up with multiple dashboards at sort order `4`, for example)
5. There's no semi-colon on the end here, which indicates it hasn't been linted

Reply:
I've just swapped this whole thing with a `lodash.orderby` because as I made it smarter I realised I was reinventing the wheel.